nixos/modules/services/continuous-integration/jenkins/default.nix

```diff
@@ -89,11 +83,13 @@ in
 
       package = lib.mkPackageOption pkgs "jenkins" { };
 
+      javaPackage = lib.mkPackageOption pkgs "jdk21" { };
+
       packages = lib.mkOption {
         default = [
           pkgs.stdenv
           pkgs.git
-          pkgs.jdk17
+          pkgs.jdk21
```

Comment:
i am no expert here. but does it make sense to use the `cfg.javaPackage` ?

Reply:
My personal opinion is that the built-in build node shouldn't do much and should be kept dumb as possible. So I actually would move the default config to an example config and keep the defaults empty, but that's something to discuss in another PR. :)